src/nodejs/neon-js/cli/rpc/getBlock.js

```diff
@@ -23,43 +28,63 @@ program
   .version('0.2.0')
   .usage('')
   .option('-d, --debug', 'Debug')
-  .option('-n, --node [node]', 'set RPC node to use (be sure to preface with https://)')
+  .option('-n, --node [node]', 'set RPC node to use (be sure to preface with https://), if not provided will try to use node with tallest block')
   .option('-h, --hash [hash]', 'specify the hash of the block to fetch, if no hash or index is supplied will get the tallest')
   .option('-i, --index [index]', 'specify the number of the block to fetch, if no hash or index is supplied will get the tallest')
   .option('-t, --time', 'Only return time field of last block')
   .option('-H, --Human', 'I am human so make outputs easy for human')
   .option('-N, --Net [Net]', 'Select network [net]: i.e., TestNet or MainNet', 'TestNet')
-  // TODO move all -n args to -N for network
   .parse(process.argv);
 
 if (program.debug) {
   print('DEBUGGING: ' + __filename)
   defly = true
+  netutil.debug()
 }
 
 if (!program.node) {
   // get a node from the list and try it
   let net = netutil.resolveNetworkId(program.Net)
 
-  dbg.logDeep('nets: ', cfg.get_nodes(net))
+  nodes = cfg.get_nodes(net)
 
-} else node = program.node
+  if (defly) dbg.logDeep('config nodes: ', nodes)
 
-if (program.hash) arg = program.hash
-if (program.index) arg = parseInt(program.index)
+  // TODO: look at if dynamic node selection should happen here or in the module
+
+  netutil.getNodesByTallest(nodes).then(rankedNodes => {
+    if (defly) dbg.logDeep('sorted nodes: ', rankedNodes)
+    nodes = rankedNodes
+    getBlockWrapper(nodes)
+  }).catch (error => {
+      console.log('neon-js.getNodesByTallest(): ' + error.message)
+  })
 
-let runtimeArgs = {
-  'debug': defly,
-  'node': node,
-  'hash': program.hash,
-  'index': program.index,
-  'time': program.time ? program.time : false,
-  'human': program.Human ? program.Human : false,
-  'index': program.index
+} else {
+  nodes.push({ "url": program.node })
+  getBlockWrapper(nodes)
 }
 
-if (defly) dbg.logDeep('runtimeArgs: ', runtimeArgs)
+if (program.hash) arg = program.hash
+if (program.index) arg = parseInt(program.index)
 
-getBlock.run(runtimeArgs).then((r) => {
-  dbg.logDeep('\nresult:\n', r)
-})
+function getBlockWrapper(nodelist) {
+  let runtimeArgs = {
+    'debug': defly,
+    'node': nodelist[0].url,
```

Comment:
same here, do we want to keep this hardcoded or dynamic?

Reply:
For now hardcoded since the default is to get the tallest node if a node isn't provided on CLI.